### PR TITLE
Week 5

### PR DESCRIPTION
**Pull request by Erkan-5Minds**

### CODE REVIEW

File: Week_5/inheritance/inheritance.c [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+char random_allele();
+
+int main(void)
+{
+    srand(time(0));
+    person *p = create_family(GENERATIONS);
+
+    print_family(p, 0);
+    free_family(p);
+}
```
Comment (stefanluth) [inline Week_5/inheritance/inheritance.c:28]: missing return

File: Week_5/inheritance/inheritance.c [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+{
+    if (p == NULL){
+        return;
+    }
+    
+    free_family(p->parents[0]);
+    free_family(p->parents[1]);
+
+    free(p);
+}
```
Comment (stefanluth) [inline Week_5/inheritance/inheritance.c:69]: missing return

File: Week_5/inheritance/inheritance.c [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+    for (int i = 0; i < generation * INDENT_LENGTH; i++)
+    {
+        printf(" ");
+    }
+
+    printf("Generation %i, blood type %c%c\n", generation, p->alleles[0], p->alleles[1]);
+    print_family(p->parents[0], generation + 1);
+    print_family(p->parents[1], generation + 1);
+}
```
Comment (stefanluth) [inline Week_5/inheritance/inheritance.c:86]: missing return

File: Week_5/speller/dictionary.c [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+#include "dictionary.h"
+
+typedef struct node
+{
+    char word[LENGTH + 1];
+    struct node *next;
+}
+node;
+const unsigned int N = 500;
```
Comment (stefanluth) [inline Week_5/speller/dictionary.c:19]:
```suggestion
} node;

const unsigned int N = 500;
```

I would prefer this style :)

File: Week_5/speller/dictionary.c [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+    
+    while (cursor != NULL)
+    {
+        if(strcasecmp(cursor -> word, word) == 0)
+        {
+            return true;
+        }
+        cursor = cursor -> next;
+    }
+    return false;
```
Comment (stefanluth) [inline Week_5/speller/dictionary.c:37]:
```suggestion
    }

    return false;
```

File: Week_5/speller/dictionary.c [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+unsigned int hash(const char *word)
+{
+    unsigned int value = 0;
+    unsigned int key_len = strlen(word);
+    for (int i = 0; i < key_len; i++)
+    {
+        value = value + 37 * tolower(word[i]);
+    }
+    value = value % N;
+    return value;
```
Comment (stefanluth) [inline Week_5/speller/dictionary.c:49]:
```suggestion
    value = value % N;

    return value;
```

File: Week_5/speller/dictionary.c [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+bool load(const char *dictionary)
+{
+    FILE *open_dictionary = fopen(dictionary,"r");
+    if (open_dictionary == NULL)
+    {
+        return false;
+    }
+    char Dword[LENGTH + 1];
+    while(fscanf(open_dictionary,"%s", Dword) != EOF)
+    {
```
Comment (stefanluth) [inline Week_5/speller/dictionary.c:61]:
```suggestion
    }

    char Dword[LENGTH + 1];

    while(fscanf(open_dictionary,"%s", Dword) != EOF)
    {
```

Comment (stefanluth) [reply Week_5/speller/dictionary.c:61]: Add some blank lines in this file to improve readability :)

File: Week_5/speller/dictionary.c [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+{
+    for (int i = 0; i < N; i++)
+    while (table[i] != NULL)
+    {
+        node *tmp = table[i] ->next;
+        free(table[i]);
+        table[i] = tmp;
+    }
+    return true;
+}
```
Comment (stefanluth) [inline Week_5/speller/dictionary.c:101]: This function always returns true

File: Week_5/speller/speller.c [hunk 9]
```diff
@@ -0,0 +1,196 @@
...
+#undef calculate
+#undef getrusage
+
+// Default dictionary
+#define DICTIONARY "dictionaries/large"
+
+// Prototype
+double calculate(const struct rusage *b, const struct rusage *a);
+
+int main(int argc, char *argv[])
```
Comment (stefanluth) [inline Week_5/speller/speller.c:20]: This is one huge function. Extract some code :)

**Review by stefanluth — APPROVED** [review]

**Review by DevBastiS — APPROVED** [review]
What Stefan said after that you can merge

**Comment by stefanluth** [comment]
Merged without changes? :) @Erkan-5Minds @DevBastiS

**Pull request merged**